mlflow/pyfunc/__init__.py

```diff
@@ -716,6 +719,64 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl)
 
 
+def _download_model_conda_env(model_uri):
+    model_meta_path = _download_artifact_from_uri(os.path.join(model_uri, MLMODEL_FILE_NAME))
+    model_meta = Model.load(model_meta_path)
+    conda_yml_file_name = model_meta.flavors[FLAVOR_NAME][ENV]
```

Comment:
`mlflow.utils.model_utils._get_flavor_configuration_from_uri` (https://github.com/mlflow/mlflow/blob/01182994ec31966dbe8528603d9e1a0ed2188234/mlflow/utils/model_utils.py#L45) will take care of getting the flavor configuration for us. It will also throw a descriptive error if the model doesn't have the pyfunc flavor. Let's use it.